backbone.js

```diff
@@ -106,49 +97,19 @@
     // callbacks for the event. If `name` is null, removes all bound
     // callbacks for all events.
     off: function(name, callback, context) {
-      if (!this._events || !eventsApi(this, 'off', name, [callback, context])) return this;
-
-      // Remove all callbacks for all events.
-      if (!name && !callback && !context) {
-        this._events = void 0;
-        return this;
-      }
-
-      var names = name ? [name] : _.keys(this._events);
-      for (var i = 0, length = names.length; i < length; i++) {
-        name = names[i];
-
-        // Bail out if there are no events stored.
-        var events = this._events[name];
-        if (!events) continue;
-
-        // Remove all callbacks for this event.
-        if (!callback && !context) {
-          delete this._events[name];
-          continue;
-        }
-
-        // Find any remaining events.
-        var remaining = [];
-        for (var j = 0, k = events.length; j < k; j++) {
-          var event = events[j];
-          if (
-            callback && callback !== event.callback &&
-            callback !== event.callback._callback ||
-            context && context !== event.context
-          ) {
-            remaining.push(event);
-          }
-        }
-
-        // Replace events if there are any remaining.  Otherwise, clean up.
-        if (remaining.length) {
-          this._events[name] = remaining;
-        } else {
-          delete this._events[name];
+      if (!this._events) return this;
+      this._events = eventsApi(offApi, this._events, name, callback, context);
+
+      var listeners = this._listeners;
+      if (listeners) {
+        var ids = context != null ? [context._listenId] : _.keys(listeners);
+        for (var i = 0, length = ids.length; i < length; i++) {
+          var listener = listeners[ids[i]];
+          if (!listener) break;
```

Comment:
When you `listenTo`, context is forced to be `this`. If we're `off`ing, we need to consider two circumstances:

1. No context
    - We'll loop through all our listeners, since any could be affected.
    - This conditional will never be true, so we'll never `break`.
2. Context
    - Only loop through the listeners of the corresponding `_listenId`, since **only** that listener will have the same `context`
    - If we have `listeners`, but not that `_listenId`, then `break`.
      - `ids` is guaranteed to only equal `[context._listenId]`, so there's no use `continue`ing.
    - If we have that `_listenId`, all is good.